src/firefly/java/edu/caltech/ipac/firefly/server/query/ObsCorePackager.java

```diff
@@ -136,37 +136,19 @@ private List<FileGroup> computeFileGroup(DownloadRequest request) throws DataAcc
                 String access_format = (String) dgDataUrl.get(idx, ACCESS_FORMAT);
                 URL url = new URL(access_url);
 
-                String colNames = request.getSearchRequest().getParam(TEMPLATE_COL_NAMES);
-                boolean useSourceUrlFileName= request.getSearchRequest().getBooleanParam(USE_SOURCE_FILE_NAME,false);
-                String[] cols = colNames != null ? colNames.split(",") : null;
-                String ext_file_name = getFileName(idx,cols,useSourceUrlFileName,dgDataUrl,url);
+                String zipType = request.getParam(ZIP_TYPE); //flat or folder
+                String extName = "folder".equalsIgnoreCase(zipType) ? "row_" + (idx+1) + "/" : null; //if flattened, no external name, just set null
 
-                if (access_format != null) {
-                    if (access_format.contains(DATALINK)) {
-                        ext_file_name = getUniqueFileNameForGroup(ext_file_name, dataLinkFolders);
-                        List<FileInfo> tmpFileInfos = parseDatalink(url, request, dgDataUrl, idx, ext_file_name);
-                        fileInfos.addAll(tmpFileInfos);
-                    }
-                    else { //non datalink entry -  such as fits,jpg etc.
-                        if (!useSourceUrlFileName || FileUtil.getExtension(ext_file_name).isEmpty()) {
-                            String extension = access_format.replaceAll(".*/", "");
-                            ext_file_name += "." + extension;
-                        }
-                        FileInfo fileInfo = new FileInfo(access_url, ext_file_name, 0);
-                        fileInfos.add(fileInfo);
-                    }
-                }
-                else {
-                    if (datalinkServDesc != null) { //parse the datalink url found in the non-obscore table
-                        List<FileInfo> tmpFileInfos = parseDatalink(url, request, dgDataUrl, idx, ext_file_name);
-                        fileInfos.addAll(tmpFileInfos);
-                    }
-                    else { //access_format & access_url are both null, so try and get it from the url's Content_Type
-                        FileInfo fileInfo = new FileInfo(access_url, ext_file_name, 0);
-                        fileInfos.add(fileInfo);
-                    }
-                }
+                boolean isDatalink = (access_format != null && access_format.contains(DATALINK)) ||
+                        (access_format == null && datalinkServDesc != null);
```

Comment:
This simplifies to : datalinkServDesc != null || access_format != null && access_format.contains(DATALINK).
Is that the logic you intended?

Reply:
Yes, I suppose that would work as well. The second condition there `datalinkServDesc != null` is essentially checking for non-obscore tables containing datalink (like `mer-catalog` in Euclid, for instance). So I'll make the change to simplify this.